### DISCUSSION TITLE
POST and PUT request dont work when using javascript fetch

### DISCUSSION THREAD

**Discussion by GHF-Hiricis** [fastapi/fastapi#12778] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/newsletterContentUpdate/{newsletter_content_id}",response_model=NewsletterContentUpdates)
async def update_newsletter_content(newsletter_content_id: int, content: NewsletterContentUpdates, db: db_dependency):
    newsletter_content = db.query(models.NewsletterContent).filter(models.NewsletterContent.id == newsletter_content_id).first()
    
    if not newsletter_content:
        raise HTTPException(status_code=404, details="No content found")

    newsletter_content.used_ai = content.used_ai if content.used_ai is not None else newsletter_content.used_ai
    newsletter_content.layout = content.layout if content.layout is not None else newsletter_content.layout
    newsletter_content.sumary = content.sumary if content.sumary is not None else newsletter_content.sumary
    newsletter_content.status = content.status if content.status is not None else newsletter_content.status
    
    db.commit()
    db.refresh(newsletter_content)
    return newsletter_content
```

### Description

When i use the docs page and postman the it works but when i try and use the javascript fetch i get this error

INFO:     127.0.0.1:51770 - "OPTIONS /newsletterContentUpdate/1/ HTTP/1.1" 405 Method Not Allowed

this is my javascript fetch code:

      await fetch("http://127.0.0.1:8000/newsletterContentUpdate/1", {
        method: "POST",
        headers: {
          accept: "application/json",
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          used_ai: 0,
          layout: "layout-1",
          sumary: "string",
          status: 0,
        }),
      });

### Operating System

Windows

### FastAPI Version

0.115.4

### Pydantic Version

2.9.2

### Python Version

3.12.7

**Comment by luzzodev** [comment, marked as answer]
Hello i think you have to configure CORS properly.
You can read more about CORS and Fastapi on [CORS TUTORIAL](https://fastapi.tiangolo.com/tutorial/cors/#steps)

**Reply by GHF-Hiricis** [reply]
tha was it, thanks
